pkg/utils/watchJob.go

```diff
@@ -99,16 +101,24 @@ func (engineDetails EngineDetails) DeleteJobAccordingToJobCleanUpPolicy(experime
 		return "", err
 	}
 
-	if expEngine.Spec.JobCleanUpPolicy == v1alpha1.CleanUpPolicyDelete || string(expEngine.Spec.JobCleanUpPolicy) == "" {
-		log.Infof("deleting the job as jobCleanPolicy is set to %v", expEngine.Spec.JobCleanUpPolicy)
+	jobCleanUpPolicy := v1alpha1.CleanUpPolicyRetain
+	if expEngine.Spec.JobCleanUpPolicy != "" {
+		jobCleanUpPolicy = expEngine.Spec.JobCleanUpPolicy
+	}
 
+	switch jobCleanUpPolicy {
+	case v1alpha1.CleanUpPolicyDelete:
+		log.Infof("deleting the job as jobCleanPolicy is set to %v", jobCleanUpPolicy)
 		deletePolicy := metav1.DeletePropagationForeground
-		deleteJob := clients.KubeClient.BatchV1().Jobs(experiment.Namespace).Delete(experiment.JobName, &metav1.DeleteOptions{
+		if deleteJob := clients.KubeClient.BatchV1().Jobs(experiment.Namespace).Delete(experiment.JobName, &metav1.DeleteOptions{
 			PropagationPolicy: &deletePolicy,
-		})
-		if deleteJob != nil {
+		}); deleteJob != nil {
 			return "", errors.Errorf("unable to delete ChaosExperiment Job name: %v, in namespace: %v, error: %v", experiment.JobName, experiment.Namespace, err)
 		}
+	case v1alpha1.CleanUpPolicyRetain:
```

Comment:
lets print some log when the `deleteJob == nil`

Reply:
line 111 already printed for the positive case